### ISSUE TITLE
Global options

### ISSUE THREAD

**Issue by memelet** [silas/node-consul#7]
In a little spike I have:

```
consul = require("consul")({
  host: "s-consul-server-01"
})

consul.kv.get {key: "tenant/t1", dc: "production"}, (err, r) ->
  console.log "--- err=", err
  console.log "--- r=", r
```

This works and returns the kv value.

But when I try to move the dc to the global options:

```
consul = require("consul")({
  host: "s-consul-server-01"
  dc: "production"
})

consul.kv.get {key: "tenant/t1"}, (err, r) ->
  console.log "--- err=", err
  console.log "--- r=", r
```

Here the callback is invoked with undefined for both 'err' and 'r'.

From the docs it looks like I should be able to do this. Am I missing something here?

**Comment by silas** [comment]
Yeah, the common options section should be more explicit, it basically works for everything except the client constructor.

**Comment by memelet** [comment]
So there is no way to globally define the consul client url to use at the client level? It must be passed to each function/query?

**Comment by silas** [comment]
You can define all these options at the client level: https://github.com/silas/node-consul#init

All these options must be defined at the method level: https://github.com/silas/node-consul#common-options

So you can define a host, but not a dc, at the client level.

I expect the reason you're getting undefined in your second example is because the local (default) dc doesn't have the key set.

You could confirm this by printing the actual res:

``` coffee
consul = require("consul")({
  host: "s-consul-server-01"
})

consul.kv.get {key: "tenant/t1"}, (err, r, res) ->
  console.log "--- err=", err
  console.log "--- r=", r
  console.log "--- res=", res.statusCode, res.headers, res.body
```

**Comment by ekristen** [comment]
It would be really nice to be able to define default options at the client level, especially things like the acl token.

**Comment by silas** [comment]
Released to npm in version `>= 0.16.0`.

Example

```
var consul = consul({ defaults: { token: '123' } });
```

You can set these defaults for now: https://github.com/silas/node-consul/blob/e0d183530818af79dd65c498edc4146087839648/lib/constants.js#L11-L19
